src/server/game/Spells/Auras/SpellAuraEffects.cpp

```diff
@@ -1575,6 +1575,8 @@ void AuraEffect::HandleAuraGhost(AuraApplication const* aurApp, uint8 mode, bool
         target->SetFlag(PLAYER_FLAGS, PLAYER_FLAGS_GHOST);
         target->m_serverSideVisibility.SetValue(SERVERSIDE_VISIBILITY_GHOST, GHOST_VISIBILITY_GHOST);
         target->m_serverSideVisibilityDetect.SetValue(SERVERSIDE_VISIBILITY_GHOST, GHOST_VISIBILITY_GHOST);
+        target->SetWaterWalking(true);
+        target->SetRooted(false);
     }
```

Comment:
should SetCanTransitionBetweenFlyAndSwim(true) added here as well?

I believe i saw this packet sent on newly death players